src/store/actions/performNextAction/swap.js

```diff
@@ -72,11 +72,32 @@ async function initiateSwap ({ getters, dispatch }, { order, network, walletId }
   }
 }
 
+async function fundSwap ({ getters, dispatch }, { order, network, walletId }) {
+  if (await dispatch('checkIfQuoteExpired', { network, walletId, order })) return
+
+  const toClient = getters.client(network, walletId, order.to)
+
+  const fundTx = await toClient.swap.fundSwap(
+    order.fromFundHash,
+    order.fromAmount,
+    order.fromCounterPartyAddress,
+    order.fromAddress,
+    order.secretHash,
+    order.swapExpiration,
+    order.fee
+  )
+
+  return {
+    fundTxHash: fundTx?.hash,
```

Comment:
Should also commit the full object like the other transactions
for ethereum full object needs to be passed to `updateTransactionFee` to ensure it can do this

Reply:
also should name this consistent to others 
`fromFundTx` already exists for intiiation
`fromSecondaryFundTx` is what you would call it to be consistent with how the agent calls it